### ISSUE TITLE
Disambiguating version control system and repository

### ISSUE THREAD

**Issue by adityasaky** [ossf/security-baseline#66]
In OSPS-01:

> The project's [version control system](https://github.com/ossf/security-baseline/blob/main/baseline.md#version-control-system) MUST require [multi-factor authentication](https://github.com/ossf/security-baseline/blob/main/baseline.md#multi-factor-authentication) for [collaborators](https://github.com/ossf/security-baseline/blob/main/baseline.md#collaborator) modifying the project [repository](https://github.com/ossf/security-baseline/blob/main/baseline.md#repository) settings or accessing sensitive data.

The definition for VCS points to examples like Git that cannot enforce multi-factor authentication. The definition for "repository" is closer to a source control platform (SCP) such as GitHub or GitLab that can enforce multi-factor authentication. I'm unsure if just replacing VCS with repository is sufficient, especially since the word "repository" doesn't always mean the centrally hosted copy to everyone.

Separately, does this requirement exclude other options for hosting source code? For example, I'm unsure if https://git.kernel.org/ can meet the MFA requirement.

**Comment by adityasaky** [comment]
>This has been a point of contention for us on other entries as well. The current stance is that if a project or tool doesn't meet the baseline, it's their prerogative to either explain the reasoning to their users, or add features to conform as needed. (We shouldn't lower the bar simply because someone isn't currently passing it.)

Overall, I agree with not lowering the bar because someone isn't currently passing it. But I think there's a question of applicability here which goes back to the definitions and desired properties for the control. Perhaps we first have to establish whether git.kernel.org even fits the SCP / VCS platform definition? It's possible it doesn't.

"Require multi-factor authentication for the project's version control system, requiring collaborators to provide a second form of authentication when accessing sensitive data or modifying repository settings." I parse this to mean git.kernel.org doesn't meet the definition, but at first read it would seem that the control does apply and the project doesn't meet it. Maybe the controls could emphasize the objective more and indicate that MFA is one way to meet the objective for SCPs like GitHub?

**Comment by funnelfiasco** [comment]
The relevant [Best Practices Badge criterion](https://www.bestpractices.dev/en/criteria#2.require_2FA) says, in part:

> The project MUST require two-factor authentication (2FA) for developers for changing a central repository or accessing sensitive data

So if we rewrote OSPS-AC-01.01 to:

> When a user attempts to access a sensitive resource in the project's central repository, the system MUST require the user to complete a multi-factor authentication process.

(This is `s/version control system/central repository/`)

Would that sufficiently address your concerns, @adityasaky?

**Comment by david-a-wheeler** [comment]
What does "access" mean? Do you mean "read or change"? I suggest changing "access" to "read or change" as I find "access" is often dreadfully ambiguous.

**Comment by adityasaky** [comment]
> (This is `s/version control system/central repository/`)

I agree with @david-a-wheeler's point. I wonder if we can qualify it as "if the central repository supports making changes, the system must require the user to complete a multi-factor authentication process". This addresses scenarios with lightweight git front ends I think?

**Comment by funnelfiasco** [comment]
Opened #358 with my proposed text to try to resolve this issue. I don't see a need to qualify it with "if the central repository supports" because if it doesn't support that, the user can't try to read or modify the sensitive resource.
